redux-form/index.d.ts

```diff
@@ -1,476 +1,403 @@
+        registeredFields: React.Validator<any>;
+        touchOnBlur: React.Validator<boolean>;
+        touchOnChange: React.Validator<boolean>;
+        triggerSubmit: React.Validator<boolean>;
+        validate: React.Validator<Fn>;
+        warn: React.Validator<Fn>;
+    }
+    const propTypes: PropTypes;
+
+    /* SubmissionError */
+    class SubmissionError<FormData extends DataShape> extends Error {
+        constructor(errors?: FormErrors<FormData>)
+    }
+
+    /* Action Creators */
+    function arrayInsert(form: string, field: string, index: number, value: any): Redux.Action;
+    function arrayMove(form: string, field: string, from: number, to: number): Redux.Action;
+    function arrayPop(form: string, field: string): Redux.Action;
+    function arrayPush(form: string, field: string, value: any): Redux.Action;
+    function arrayRemove(form: string, field: string, index: number): Redux.Action;
+    function arrayRemoveAll(form: string, field: string): Redux.Action;
+    function arrayShift(form: string, field: string): Redux.Action;
+    function arraySplice(form: string, field: string, index: number, removeNum: number, value: any): Redux.Action;
+    function arraySwap(form: string, field: string, indexA: number, indexB: number): Redux.Action;
+    function arrayUnshift(form: string, field: string, value: any): Redux.Action;
+    function autofill(form: string, field: string, value: any): Redux.Action;
+    function blur(form: string, field: string, value: any): Redux.Action;
+    function change(form: string, field: string, value: any): Redux.Action;
+    function destroy(form: string): Redux.Action;
+    function focus(form: string, field: string): Redux.Action;
+    function initialize<FormData extends DataShape>(form: string, data: FormData, keepDirty?: boolean): Redux.Action;
+    function reset(form: string): Redux.Action;
+    function startAsyncValidation(form: string): Redux.Action;
+    function startSubmit(form: string): Redux.Action;
+    function stopSubmit<FormData extends DataShape>(form: string, errors?: FormErrors<FormData>): Redux.Action;
+    function stopAsyncValidation<FormData extends DataShape>(form: string, errors?: FormErrors<FormData>): Redux.Action;
+    function submit(form: string): Redux.Action;
+    function touch(form: string, ...fields: string[]): Redux.Action;
+    function untouch(form: string, ...fields: string[]): Redux.Action;
+
+    /* Selectors */
+    type DataSelector = (form: string) => (state: any) => any;
+    type ErrorsSelector = (form: string) => (state: any) => FormErrors<any>;
+    type BooleanSelector = (form: string) => (state: any) => boolean;
+
+    const getFormValues: DataSelector;
+    const getFormSyncErrors: ErrorsSelector;
+    const getFormSubmitErrors: ErrorsSelector;
+    const isDirty: BooleanSelector;
+    const isPristine: BooleanSelector;
+    const isValid: BooleanSelector;
+    const isInvalid: BooleanSelector;
 }
 
+export as namespace ReduxForm
```

Comment:
You don't have to declare a root namespace when you use `export as namespace`. Just use regular `export`s on top level